flutter/lib/src/sentry_flutter_options.dart

```diff
@@ -385,11 +398,33 @@ class SentryFlutterOptions extends SentryOptions {
 class _SentryFlutterExperimentalOptions {
   /// Replay recording configuration.
   final replay = SentryReplayOptions();
-}
 
-/// Callback being executed in [ScreenshotEventProcessor], deciding if a
-/// screenshot should be recorded and attached.
-typedef BeforeScreenshotCallback = FutureOr<bool> Function(
-  SentryEvent event, {
-  Hint? hint,
-});
+  /// Privacy configuration for masking sensitive data in the Screenshot and Session Replay.
+  /// Screen content masking redaction is:
+  /// - enabled by default for SessionReplay
+  /// - disabled by default for screenshots captured with events.
+  /// In order to redact screenshots captured with events, access or change
+  /// this property in your application: `options.experimental.privacy`.
+  /// Doing so will indicate that you want to configure privacy settings and
+  /// will enable screenshot redaction alongside the default replay redaction.
+  /// Note: this will change in a future SDK major release to enable screenshot
+  /// redaction by default for all captures.
+  SentryPrivacyOptions get privacy {
+    // If the user explicitly sets the privacy setting, we use that.
+    // Otherwise, we use the default settings, which is no redaction for screenshots
+    // and full redaction for session replay.
+    // This property must only by accessed by user code otherwise it defeats the purpose.
+    _privacy ??= SentryPrivacyOptions();
+    return _privacy!;
+  }
```

Comment:
@vaind what will/should happen, if the user disables all `privacy` settings? Will this also disable masking for `Session Replay`? Or will this only affect the `Screenshot` part?

Reply:
> Will this also disable masking for Session Replay

yes